app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/CategoryTree.php

```diff
@@ -84,6 +85,10 @@ public function __construct(
     public function getTree(ResolveInfo $resolveInfo, int $rootCategoryId): \Iterator
     {
         $categoryQuery = $resolveInfo->fieldNodes[0];
+        if ($this->isCategoryActive($rootCategoryId) === false) {
```

Comment:
Please cover this case:
category with "path": "1/2/3/4/5" has products assigned and is enabled
category with "path": "1/2/3/4" has no products assigned and is enabled
category with "path": "1/2/3" has no products assigned and is disabled

Assert 1: for category 3 product_count throws GraphQlNoSuchEntityException
Assert 2: for category 5 product_count throws GraphQlNoSuchEntityException
Assert 2 will fail.